Gordon360/Services/RecIM/MatchService.cs

```diff
@@ -0,0 +1,217 @@
+                                    .Join(_context.MatchTeam.Where(q => q.TeamID == mt.TeamID).Join(
+                                            _context.MatchTeam,
+                                            mt0 => mt0.MatchID,
+                                            mt1 => mt1.MatchID,
+                                            (mt0, mt1) => new
+                                            {
+                                                MatchID = mt0.MatchID,
+                                                OwnScore = mt0.Score,
+                                                OpposingID = mt1.TeamID,
+                                                OpposingScore = mt1.Score,
+                                                Status = mt0.Score > mt1.Score
+                                                        ? "Win"
+                                                        : mt0.Score < mt1.Score
+                                                            ? "Lose"
+                                                            : "Tie"
+                                            }),
+                                        match => match.ID,
+                                        matchTeamJoin => matchTeamJoin.MatchID,
+                                        (match, matchTeamJoin) => new TeamMatchHistoryViewModel
+                                        {
+                                            opponent = _context.Team.Where(t => t.ID == matchTeamJoin.OpposingID)
+                                                .Select(o => new TeamViewModel
+                                                {
+                                                    ID = o.ID,
+                                                    Name = o.Name,
+                                                    Logo = o.Logo
+                                                }).FirstOrDefault(),
+                                            OwnScore = matchTeamJoin.OwnScore,
+                                            OpposingScore = matchTeamJoin.OpposingScore,
+                                            Status = matchTeamJoin.Status,
+                                            Time = match.Time
+                                        })
+                                    .OrderByDescending(history => history.Time)
+                                    .Take(5),
+                                TeamRecord = _context.SeriesTeam
+                                           .Where(st => st.SeriesID == m.SeriesID && st.TeamID == mt.TeamID)
+                                           .Select(st => new TeamRecordViewModel
+                                           {
+                                               Win = st.Win,
+                                               Loss = st.Loss ?? 0,
+                                           })
+                            })
+                        }).FirstOrDefault();
```

Comment:
use an implicit operator or split this logic out into another method. This is far too complex for a single line action.

Also, `FirstOrDefault` can take a lambda function, so you can just replace `.Select(...` with that

Reply:
As stated, the team thing will be handled by TeamService once that is created and I can then use _teamService.GetTeamByMatchID or something along those lines. 

I think you're thinking of replacing .Where with .FirstOrDefault. I don't think it works for select. Though I can't actually use .FirstOrDefault in replacement for the first .Where because you can't actually Select from a FirstOrDefault

Reply:
No, you're right on the .FirstOrDefault. I misread because the select is so long :)